napari/components/_viewer/view/main.py

```diff
@@ -214,20 +218,44 @@ def on_mouse_release(self, event):
     def on_key_press(self, event):
         """Called whenever key pressed in canvas.
         """
-        if (
-            event.text in self.viewer.key_bindings
-            and not event.native.isAutoRepeat()
-        ):
-            self.viewer.key_bindings[event.text](self.viewer)
+        if event.native.isAutoRepeat() or event.key is None:
             return
 
+        comb = components_to_comb(event.key.name, event.modifiers)
+
         layer = self.viewer.active_layer
+
+        # TODO: remove me once keybinding system converted
         if layer is not None:
             layer.on_key_press(event)
 
+        if layer is not None and comb in layer.keymap:
```

Comment:
Could you expand the names of things? `comb` -> `key_combo`, `parent` -> `key_receiver` (or somesuch, something more descriptive of what it is), `func` -> `function`, `gen` -> `result`?

Reply:
sure, but do you really think `function` is that much more descriptive than `func`? I thought that's pretty standard

I also chose `gen` because sometimes it returns a generator. still want to go with `result` or can you think of something that better captures this?

Reply:
@kne42 func is the least worrisome. `gen` being *sometimes* a generator is a huge problem! LOL `value_or_generator` is the only thing I can think of otherwise, but `result` works in any case. maybe add a comment, "can be a single value or a generator"?